src/Service/MigrationService.php

```diff
@@ -0,0 +1,638 @@
+     */
+    public const BASE_MIGRATION = 'm000000_000000_base';
+
+    private string $comment = '';
+    private bool $compact = false;
+    private array $fields = [];
+    private array $generatorTemplateFiles = [
+        'create' => '@views/migration.php',
+        'table' => '@views/createTableMigration.php',
+        'dropTable' => '@views/dropTableMigration.php',
+        'addColumn' => '@views/addColumnMigration.php',
+        'dropColumn' => '@views/dropColumnMigration.php',
+        'junction' => '@views/createTableMigration.php',
+    ];
+    private int $maxNameLength = 180;
+    private ?int $migrationNameLimit = null;
+    private array $migrationNamespaces = [];
+    private array $migrationPath = ['@migration'];
+    private string $migrationTable = '{{%migration}}';
+    private string $templateFile = '@views/migration.php';
+    private bool $useTablePrefix = true;
+    private string $version = '1.0';
+    private Aliases $aliases;
+    private Connection $db;
+    private ConsoleHelper $consoleHelper;
+    private Inflector $inflector;
+
+    public function __construct(Aliases $aliases, Connection $db, ConsoleHelper $consoleHelper)
+    {
+        $this->aliases = $aliases;
+        $this->db = $db;
+        $this->consoleHelper = $consoleHelper;
+        $this->inflector = new Inflector();
+    }
+
+    /**
+     * This method is invoked right before an action is to be executed (after all possible filters.)
+     *
+     * It checks the existence of the {@see migrationPath}.
+     *
+     * @throws InvalidConfigException if directory specified in migrationPath doesn't exist and action isn't "create".
+     *
+     * @return bool whether the action should continue to be executed.
+     */
+    public function before($defaultName): bool
+    {
+        if (empty($this->migrationNamespaces) && empty($this->migrationPath)) {
+            throw new InvalidConfigException(
+                'At least one of `migrationPath` or `migrationNamespaces` should be specified.'
+            );
+        }
+
+        foreach ($this->migrationNamespaces as $key => $value) {
+            $this->migrationNamespaces[$key] = trim($value, '\\');
+        }
+
+        if (is_array($this->migrationPath)) {
+            foreach ($this->migrationPath as $i => $path) {
+                $this->migrationPath[$i] = $this->aliases->get($path);
+            }
+        } elseif ($this->migrationPath !== null) {
+            $path = $this->aliases->get($this->migrationPath);
+            if (!is_dir($path)) {
+                if ($defaultName !== 'migrate/create') {
+                    throw new InvalidConfigException(
+                        "Migration failed. Directory specified in migrationPath doesn't exist: {$this->migrationPath}"
+                    );
+                }
+                @FileHelper::createDirectory($path);
+            }
+            $this->migrationPath = $path;
```

Comment:
Looks like a bug. `$migrationPath` is typed as array.

Reply:
Fixed.